core-services/release-controller/rpms-cdn.yaml

```diff
@@ -0,0 +1,213 @@
+apiVersion: v1
+kind: List
+items:
+
+# each RPM (group,release) pair gets a service and deployment to serve RPM content from the mirrors
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: rh-cdn
+    namespace: ocp
+  spec:
+    selector:
+      app: rh-cdn
+    ports:
+    - port: 80
+      targetPort: 8080
+    type: ClusterIP
+
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: rh-cdn
+    namespace: ocp
+    labels:
+      app: rh-cdn
+    annotations:
+      image.openshift.io/triggers: '[{"from":{"kind":"ImageStreamTag","name":"content-mirror:latest","namespace":"ci"},"fieldPath":"spec.template.spec.containers[?(@.name==\"mirror\")].image"}]'
+  spec:
+    replicas: 2
+    selector:
+      matchLabels:
+        app: rh-cdn
+    template:
+      metadata:
+        labels:
+          app: rh-cdn
+      spec:
+        volumes:
+        - name: repos
+          configMap:
+            name: base-repos
+            items:
+              - key: rh-cdn-default.repo
+                path: rh-cdn-default.repo
+        - name: key
+          secret:
+            secretName: rh-cdn
+        - name: cache
+          emptyDir: {}
+        - name: run
+          emptyDir: {}
+        containers:
+        - name: mirror
+          ports:
+            - name: http
+              containerPort: 8080
+          volumeMounts:
+          - name: repos
+            mountPath: /tmp/repos
+            readOnly: true
+          - name: key
+            mountPath: /tmp/key
+            readOnly: true
+          - name: cache
+            mountPath: /tmp/cache
+          image: " "
+          workingDir: /tmp/repos
+          command:
+          - content-mirror
+          - --path=/tmp/config
+          - --max-size=5g
+          - --timeout=30m
+          - /tmp/repos
+
+# each RPM (group,release) pair gets a service and deployment to serve RPM content from the mirrors
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: rh-cdn-ppc64le
```

Comment:
Service are coupled to a version and product, not to targets.

Reply:
Now using the CDN only to supplement 4.x-rhel-8 repos. More can be migrated if this works out well.